### PR TITLE
Allow failed agent turns to be forked

### PR DESCRIPTION
**Pull request by boudra**
### Linked issue

Refs #262
Refs #1479

### Type of change

- [x] Bug fix
- [ ] New feature (with prior issue + design alignment)
- [ ] Refactor / code improvement
- [ ] Docs

### What does this PR do

Failed agent turns can now be forked even when the provider never assigned an assistant message ID. Paseo anchors the fork to the exact durable timeline position, while retaining the existing message-ID path for older hosts.

Fork context is also wrapped in a dedicated `<chat-history-summary>` block and placed before the new user prompt. Prompt assembly now has one shared ordering policy for both new agents and follow-up messages.

### How did you verify it

- Reproduced the bug with a deterministic failed provider turn: the system error rendered, but the Fork action was absent.
- Ran the same Playwright flow after the fix and observed the Fork menu open a new draft tab with the Chat history attachment.
- Ran 108 focused unit and integration tests covering live/history cursor retention, pagination merges, protocol compatibility, exact server-side slicing, XML structure, and prompt ordering.
- Ran `npm run typecheck` successfully.
- Ran `npm run lint` with zero warnings or errors.
- Ran `npm run format` and the pre-commit formatting check.

Before merge, smoke the footer action on iOS and Android. The complete failed-turn flow is covered on web by Playwright, but native menu visibility was not exercised locally.

### Risk surface

The fork request adds an optional protocol field. New clients only use cursor-only boundaries after the host advertises support; legacy message-ID forks remain unchanged. An old-client/new-host and new-client/old-host matrix was covered structurally by optional schemas and capability gating, but not by running historical binaries.

### Checklist

- [x] One focused change. Unrelated cleanups split out.
- [x] `npm run typecheck` passes
- [x] `npm run lint` passes
- [x] `npm run format` ran (Biome)
- [ ] UI changes include screenshots or video for every affected platform
- [x] Tests added or updated where it made sense

### CODE REVIEW

File: packages/server/src/server/agent/activity-curator.ts [hunk 1]
```diff
@@ -218,28 +218,46 @@ export function curateAgentActivity(
...
+      boundaryCursor: null,
       boundaryMessageId: null,
     };
   }
 
-  const boundaryIndex = input.rows.findLastIndex(
-    (row) => row.item.type === "assistant_message" && row.item.messageId === boundaryMessageId,
-  );
+  if (boundaryCursor && boundaryCursor.epoch !== input.timelineEpoch) {
+    throw new Error("Selected timeline position is no longer available.");
```
[the remark this answers is not in the document]
Comment (boudra) [reply packages/server/src/server/agent/activity-curator.ts]: Fixed in c55451234. The cursor and current epoch now travel together in a single cursorBoundary object, so a cursor-without-epoch state is no longer representable.

File: packages/app/src/agent-stream/turn-boundary.test.ts [hunk 2]
```diff
@@ -62,3 +65,22 @@ describe("resolveAssistantTurnBoundaryMessageId", () => {
...
+      resolveAssistantTurnForkBoundary({
+        items: [userMessage("user-1", 1), failedTurn],
+        startIndex: 1,
+        supportsTimelineCursor: true,
+      }),
+    ).toEqual({
+      boundaryCursor: { epoch: "timeline-1", seq: 42 },
+    });
+  });
+});
```
[the remark this answers is not in the document]
Comment (boudra) [reply packages/app/src/agent-stream/turn-boundary.test.ts:129]: Fixed in c55451234. Added direct coverage for the mixed cursor plus message-ID case, legacy hosts without cursor support, and the unavailable-boundary case.
